### ISSUE TITLE
Proguard Rules + Kotlin Serialization

### ISSUE THREAD

**Issue by ppetarr** [amplitude/experiment-android-client#51]
This is more of FYI than an open issue, but I spent too long on this problem. 

If the client doesn't use `kotlinx.serialization` then in release these types of problems will come up
```
Error decoding JSON: Serializer for class 'EvaluationFlag' is not found.
Mark the class as @Serializable or provide the serializer explicitly.
```

We don't use `kotlinx.serialization` but seeing as the library does, we needed to add proguard rules to stop the classes from being obfuscated during run-time for our release builds. You can find the [rules here](https://github.com/Kotlin/kotlinx.serialization/blob/master/rules/common.pro). 

Might be worth adding this note to your documentation + the rules for such cases.

**Comment by nico-gonzalez** [comment]
Hello! we are facing the same issue. It looks like [this change](https://github.com/amplitude/experiment-android-client/commit/b512971e725f0d53bbd4c7e14525aa8482ac06d1) introduced kotlinx.serialization but no proguard rules were added nor published by the library. I agree with @ppetarr this should either be included by the SDK itself or add the required proguard rules in the docs. Thanks!

**Comment by bgiori** [comment]
This should be fixed in `1.12.2`
